src/Microsoft.ML.Transforms/TensorflowTransform.cs

```diff
@@ -0,0 +1,501 @@
+        /// <param name="name">Name of the output column. Keep it same as in the Tensorflow model.</param>
+        /// <param name="source">Name of the input column(s). Keep it same as in the Tensorflow model.</param>
+        public TensorflowTransform(IHostEnvironment env, IDataView input, string modelFile, string name, params string[] source)
+            : this(env, new Arguments() { Column = new[] { new Column() { Source = source, Name = name } }, ModelFile = modelFile }, input)
+        {
+        }
+
+        public TensorflowTransform(IHostEnvironment env, Arguments args, IDataView input)
+            : base(env, RegistrationName, input)
+        {
+            Host.CheckValue(args, nameof(args));
+            Host.CheckUserArg(Utils.Size(args.Column) > 0, nameof(args.Column));
+            for (int i = 0; i < args.Column.Length; i++)
+                Host.CheckUserArg(Utils.Size(args.Column[i].Source) > 0, nameof(args.Column));
+
+            _batchSize = args.BatchSize;
+            _session = LoadTFSession(args.ModelFile);
+            _bindings = new Bindings(args.Column, Source.Schema, this);
+        }
+
+        private TensorflowTransform(IHost host, ModelLoadContext ctx, IDataView input)
+           : base(host, input)
+        {
+            Host.AssertValue(ctx);
+
+            _batchSize = ctx.Reader.ReadInt32();
+
+            byte[] data = null;
+            ctx.TryLoadBinaryStream("TFModel", r =>
+            {
+                data = r.ReadByteArray();
+            });
+
+            var graph = new TFGraph();
+            graph.Import(data);
+            _session = new TFSession(graph);
+            _bindings = new Bindings(ctx, Source.Schema, this);
+        }
+
+        public static TensorflowTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var h = env.Register(RegistrationName);
+            h.CheckValue(ctx, nameof(ctx));
+            h.CheckValue(input, nameof(input));
+            ctx.CheckAtModel(GetVersionInfo());
+            return h.Apply("Loading Model", ch => new TensorflowTransform(h, ctx, input));
+        }
+
+        private TFSession LoadTFSession(string modelFile)
+        {
+            var graph = new TFGraph();
+            graph.Import(File.ReadAllBytes(modelFile), "");
+            return new TFSession(graph);
+        }
+
+        public override void Save(ModelSaveContext ctx)
+        {
+            Host.AssertValue(ctx);
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            ctx.Writer.Write(_batchSize);
+
+            var buffer = new TFBuffer();
+            _session.Graph.ToGraphDef(buffer);
+
+            ctx.SaveBinaryStream("TFModel", w =>
+            {
+                w.WriteByteArray(buffer.ToArray());
+            });
```

Comment:
check for success here.

Reply:
To me comment shows at different line number after the several commits. Originally, I commented while loading at following code.

```
byte[] data = null;
ctx.TryLoadBinaryStream("TFModel", r =>
{
       data = r.ReadByteArray();
});
```

---
In reply to: [210651836](https://github.com/zeahmed/machinelearning/pull/3#discussion_r210651836) [](ancestors = 210651836,210043097)